examples/nextjs/jest.config.js

```diff
@@ -0,0 +1,34 @@
+// jest.config.js
+const nextJest = require('next/jest');
+
+const createJestConfig = nextJest({
+  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
+  dir: './',
+});
+
+// Add any custom config to be passed to Jest
+const customJestConfig = {
+  // Add more setup options before each test is run
+  // setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+  // if using TypeScript with a baseUrl set to the root directory then you need the below for alias' to work
+  moduleDirectories: ['node_modules', '<rootDir>/'],
+  testEnvironment: 'jest-environment-jsdom',
+  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+  transform: {
+    '^.+\\.css$': 'jest-preview/transforms/css',
+    '^(?!.*\\.(js|jsx|mjs|cjs|ts|tsx|css|json)$)':
+      'jest-preview/transforms/file',
+  },
+};
+
+// createJestConfig is exported this way to ensure that next/jest can load the Next.js config which is async
+module.exports = async () => {
+  const config = await createJestConfig(customJestConfig)();
+  if (typeof config === 'object') {
+    config.transformIgnorePatterns = ['/node_modules/']
```

Comment:
I think we should not overwrite the whole `transformIgnorePatterns`. Since in the future, Next can add more items to this. A safe option is something like this
```suggestion
    config.transformIgnorePatterns = config.transformIgnorePatterns.filter(pattern => pattern !== '^.+\\.module\\.(css|sass|scss)$')
```

Reply:
Agree